### PR TITLE
Implemente system of list : ZAPPY Server

### PR DESCRIPTION
**Pull request by AxelF44**
I've implemented the list system for the server !

### CODE REVIEW

File: server/include/list/list.h [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+/**
+ * @brief Delete the node of the list.
+ *
+ * @param list List where delete the node
+ * @param node Node to delete
+ */
+void list_delete(list_t *list, list_node_t *node);
+
+/**
+ * @brief Remove the first note of the list.
```
Comment (mathieurobert1) [inline server/include/list/list.h:35]:
```suggestion
 * @brief Remove the first node of the list.
```

File: server/include/list/list.h [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+ *
+ * @param list List where remove the first node
+ */
+void list_remove_front(list_t *list);
+
+/**
+ * @brief Remove the last note of the list.
```
Comment (mathieurobert1) [inline server/include/list/list.h:42]:
```suggestion
 * @brief Remove the last node of the list.
```

File: server/include/list/list.h [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+ * @param list List where remove the last node
+ */
+void list_remove_back(list_t *list);
+
+/**
+ * @brief Add a node a the back of the list.
+ *
+ * @param list List where add the new node
+ * @param data Node to add at the back
+ * @return int 0 if succes else 84
```
Comment (mathieurobert1) [inline server/include/list/list.h:53]: Why not a bool type ?

File: server/include/list/list.h [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+ */
+int list_add_back(list_t *list, node_data_t data);
+
+/**
+ * @brief Add a node a the front of the list.
+ *
+ * @param list List where add the new node
+ * @param data Node to add at the front
+ * @return int 0 if succes else 84
```
Comment (mathieurobert1) [inline server/include/list/list.h:62]: Why not a bool type ?

**Review by mathieurobert1 — CHANGES_REQUESTED** [review]
Some doxygen errors. Moreover, we can discuss about return types.

**Review by mathieurobert1 — DISMISSED** [review]
Perfect, just the chocolatine to fix.

**Review by mathieurobert1 — APPROVED** [review]
Nice good job for the tests !

**Review by Thomaltarix — APPROVED** [review]

**Review by Njord201 — APPROVED** [review]

**Pull request merged**